src/libs/blueos.ts

```diff
@@ -30,6 +51,46 @@ export const getKeyDataFromCockpitVehicleStorage = async (
   return await getBagOfHoldingFromVehicle(vehicleAddress, `cockpit/${storageKey}`)
 }
 
+export const getWidgetsFromBlueOS = async (): Promise<BlueOsWidget[]> => {
+  const vehicleStore = useMainVehicleStore()
+
+  // Wait until we have a global address
+  while (vehicleStore.globalAddress === undefined) {
+    await new Promise((r) => setTimeout(r, 1000))
+  }
+  const options = { timeout: defaultTimeout, retry: 0 }
+  const services = (await ky
+    .get(`http://${vehicleStore.globalAddress}/helper/v1.0/web_services`, options)
+    .json()) as Record<string, any>
+  // first we gather all the extra jsons with the cockpit key
+  const extraWidgets = await services.reduce(
+    async (accPromise: Promise<BlueOsWidget[]>, service: Record<string, any>) => {
+      const acc = await accPromise
+      const worksInRelativePaths = service.metadata?.works_in_relative_paths
+      if (service.metadata?.extras?.cockpit === undefined) {
+        return acc
+      }
+      const baseUrl = worksInRelativePaths
+        ? `http://${vehicleStore.globalAddress}/extensionv2/${service.metadata.sanitized_name}`
+        : `http://${vehicleStore.globalAddress}:${service.port}`
+      const fullUrl = baseUrl + service.metadata?.extras?.cockpit
+
+      const extraJson: ExtrasJson = await ky.get(fullUrl, options).json()
+      const widgets: BlueOsWidget[] = extraJson.widgets.map((widget) => {
+        return {
+          ...widget,
+          iframe_url: baseUrl + widget.iframe_url,
+          iframe_icon: baseUrl + widget.iframe_icon,
+        }
+      })
+      return acc.concat(widgets)
+    },
+    Promise.resolve([] as BlueOsWidget[])
+  )
```

Comment:
this filtering code could live on the backend under an endpoint like `get_extras(key: string) -> HttpResponse([json])` and the Cockpit would ask for the key "Cockpit".